### PR TITLE
improve gerritbot-console test tool

### PR DESCRIPTION
**Pull request by fbenkstein**
* Improve command line help messages.
* Allow simulating multiple users by not giving an email address.
* Add JSON mode.  This will be used for testing but could be used to interact with arbitrary chat systems, e.g by wrapping the executable in Python.
* Add `--quiet` mode: only error messages will be logged.
* Add hidden option to change working directory:  also useful for testing so it doesn't clobber the existing `state.json` file.

### CODE REVIEW

File: gerritbot/examples/gerritbot-console.rs [hunk 1]
```diff
@@ -5,46 +5,131 @@
...
+    #[structopt(short, long, default_value = "29418")]
     port: u32,
     /// Path to SSH private key
-    #[structopt(short = "i", parse(from_os_str))]
-    private_key_path: PathBuf,
+    #[structopt(short, long, parse(from_os_str))]
+    identity_file: PathBuf,
     /// Enable verbose output
-    #[structopt(short = "v")]
+    #[structopt(short, long)]
```
Comment (gferon) [inline gerritbot/examples/gerritbot-console.rs]: `conflicts_with` = "quiet"

File: gerritbot/examples/gerritbot-console.rs [hunk 2]
```diff
@@ -56,10 +141,19 @@ fn main() {
...
-        .verbosity(if args.verbose { 5 } else { 2 })
+        .verbosity(match (args.quiet, args.verbose) {
+            (true, _) => 0,      // ERROR
+            (false, false) => 2, // INFO
+            (_, true) => 4,      // TRACE
+        })
         .init()
         .unwrap();
 
+    if let Some(working_directory) = args.working_directory.as_ref() {
```
Comment (gferon) [inline gerritbot/examples/gerritbot-console.rs]: is it really useful? I mean, sounds a bit frivolous to me.

File: gerritbot/examples/gerritbot-console.rs (hunk 1 above)
Comment (fbenkstein) [reply gerritbot/examples/gerritbot-console.rs]: It's reflexive so the one below is enough.

File: gerritbot/examples/gerritbot-console.rs (hunk 2 above)
Comment (fbenkstein) [reply gerritbot/examples/gerritbot-console.rs]: Yes, so you can `cargo run --example gerritbot-console -- -C <tempdir>` from the behave test suite and ensure the `state.json` isn't reused.

**Pull request merged**